hibernate-core/src/test/java/org/hibernate/orm/test/mapping/lazytoone/mappedby/InverseToOneExplicitOptionTests.java

```diff
@@ -0,0 +1,239 @@
+	}
+
+	@Test
+	public void testOwnerIsProxy() {
+		inTransaction(
+				(session) -> {
+					final Customer customer = new Customer( 1, "Acme Brick" );
+					session.persist( customer );
+					final SupplementalInfo supplementalInfo = new SupplementalInfo( 1, customer, "extra details" );
+					session.persist( supplementalInfo );
+				}
+		);
+
+		sqlStatementInterceptor.clear();
+
+		final EntityPersister supplementalInfoDescriptor = sessionFactory().getMetamodel().entityPersister( SupplementalInfo.class );
+		final BytecodeEnhancementMetadata supplementalInfoEnhancementMetadata = supplementalInfoDescriptor.getBytecodeEnhancementMetadata();
+		assertThat( supplementalInfoEnhancementMetadata.isEnhancedForLazyLoading(), is( true ) );
+
+		final EntityPersister customerDescriptor = sessionFactory().getMetamodel().entityPersister( Customer.class );
+		final BytecodeEnhancementMetadata customerEnhancementMetadata = customerDescriptor.getBytecodeEnhancementMetadata();
+		assertThat( customerEnhancementMetadata.isEnhancedForLazyLoading(), is( true ) );
+
+		inTransaction(
+				(session) -> {
+
+					// 1) Get a reference to the SupplementalInfo we created
+					// 		- at that point there should be no SQL executed
+					// 2) Access the SupplementalInfo's id value
+					//		- should trigger no SQL
+					// 3) Access SupplementalInfo's `something` state
+					//		- should trigger loading the "base group" state, which only include `something`.
+					//			NOTE: `customer` is not part of this lazy group because we do not know the
+					//			Customer PK from this side
+					// 4) Access SupplementalInfo's `customer` state
+					//		- should trigger load from Customer table
+
+					final SupplementalInfo supplementalInfo = session.byId( SupplementalInfo.class ).getReference( 1 );
+
+					// we should have just the uninitialized SupplementalInfo proxy
+					//		- therefore no SQL statements should have been executed
+					assertThat( sqlStatementInterceptor.getSqlQueries().size(), is( 0 ) );
+
+					assertThat(
+							supplementalInfoEnhancementMetadata.extractLazyInterceptor( supplementalInfo ),
+							instanceOf( EnhancementAsProxyLazinessInterceptor.class )
+					);
+
+					// access the id - should do nothing with db
+					supplementalInfo.getId();
+					assertThat( sqlStatementInterceptor.getSqlQueries().size(), is( 0 ) );
+					assertThat(
+							supplementalInfoEnhancementMetadata.extractLazyInterceptor( supplementalInfo ),
+							instanceOf( EnhancementAsProxyLazinessInterceptor.class )
+					);
+
+					// this should trigger loading the entity's base state
+					supplementalInfo.getSomething();
+					assertThat( sqlStatementInterceptor.getSqlQueries().size(), is( 1 ) );
+					assertThat(
+							supplementalInfoEnhancementMetadata.extractLazyInterceptor( supplementalInfo ),
+							instanceOf( LazyAttributeLoadingInterceptor.class )
+					);
+
+					// should not trigger a load and the `customer` reference should be an uninitialized enhanced proxy
```

Comment:
Seems the comment does not reflect what is happening as you are asserting that a query is fired.

Reply:
Copy-paste error.  You can change these fwiw ;)

Reply:
You can see by the "overall expectation" comment ~ line 82 what the expectations are